apis/rabbitmq/v1beta1/rabbitmquser_webhook.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var rabbitmquserlog = logf.Log.WithName("rabbitmquser-resource")
+
+//+kubebuilder:webhook:path=/mutate-rabbitmq-openstack-org-v1beta1-rabbitmquser,mutating=true,failurePolicy=fail,sideEffects=None,groups=rabbitmq.openstack.org,resources=rabbitmqusers,verbs=create;update,versions=v1beta1,name=mrabbitmquser.kb.io,admissionReviewVersions=v1
+
+// Default implements defaulting for RabbitMQUser
+func (r *RabbitMQUser) Default(_ client.Client) {
+	rabbitmquserlog.Info("default", "name", r.Name)
+
+	// Default the username to the CR name if not specified
+	if r.Spec.Username == "" {
+		r.Spec.Username = r.Name
```

Comment:
same, can the username have any chars for being used on the MQ side, or do we have to do some validation in ValidateCreate? similar on how we do it in https://github.com/openstack-k8s-operators/infra-operator/blob/main/apis/memcached/v1beta1/memcached_webhook.go#L84 using https://github.com/openstack-k8s-operators/lib-common/blob/main/modules/common/webhook/rfc.go#L32